src/plugins/data/common/search/search_source/search_source.test.ts

```diff
@@ -836,5 +836,27 @@ describe('SearchSource', () => {
       expect(references[1].type).toEqual('index-pattern');
       expect(JSON.parse(searchSourceJSON).filter[0].meta.indexRefName).toEqual(references[1].name);
     });
+
+    test('mvt geoshape layer test', async () => {
+      // @ts-expect-error TS won't like using this field name, but technically it's possible.
+      searchSource.setField('docvalue_fields', ['prop1']);
+      searchSource.setField('source', ['geometry']);
+      searchSource.setField('fieldsFromSource', ['geometry', 'prop1']);
+      searchSource.setField('index', ({
+        ...indexPattern,
+        getSourceFiltering: () => ({ excludes: [] }),
+        getComputedFields: () => ({
+          storedFields: ['*'],
+          scriptFields: {},
+          docvalueFields: [],
+        }),
+      } as unknown) as IndexPattern);
+      const request = await searchSource.getSearchRequestBody();
+      expect(request.stored_fields).toEqual(['geometry', 'prop1']);
+      expect(request.docvalue_fields).toEqual(['prop1']);
+      expect(request._source).toEqual({
```

Comment:
Thanks for moving this into its own test case. 

After inspection, the behavior in master branch, where `_source: { includes: ['geometry', 'prop1']}` is not optimal and this PR's behavior is better. It is safe to update the expect statement to `_source: { 'geometry'}`. This PR does a better job of removing duplicate fields from _source which is great, thanks.

The existing behavior in master branch results in hits like the below. Notice how prop1's value is returned twice, once in _source and once in fields.
```
{
        "_index": "geo_shapes",
        "_id": "4",
        "_score": 0,
        "_source": {
          "prop1": 4,
          "geometry": { ...omitted }
        },
        "fields": {
          "prop1": [
            4
          ]
        }
      }
```

With this PR in place, the behavior is optimal and prop1's value is only returned a single time.
```
{
        "_index": "geo_shapes",
        "_id": "4",
        "_score": 0,
        "_source": {
          "geometry": {...omitted}
        },
        "fields": {
          "prop1": [
            4
          ]
        }
      }
```